rcl/test/rcl/test_service.cpp

```diff
@@ -76,7 +77,34 @@ class CLASSNAME (TestServiceFixture, RMW_IMPLEMENTATION) : public ::testing::Tes
   }
 };
 
-void
+rcl_ret_t
+wait_for_server_to_be_available(
+  rcl_node_t * node,
+  rcl_client_t * client,
+  size_t max_tries,
+  int64_t period_ms,
+  bool & success)
+{
+  size_t iteration = 0;
+  bool is_ready = false;
+  rcl_ret_t ret = RCL_RET_OK;
+  while (iteration < max_tries) {
+    ++iteration;
+    ret = rcl_service_server_is_available(node, client, &is_ready);
+    if (ret != RCL_RET_OK) {
+      break;
+    }
+    if (is_ready) {
+      success = true;
+      return ret;
+    }
+    std::this_thread::sleep_for(std::chrono::milliseconds(period_ms));
+  }
+  success = false;
+  return ret;
+}
+
+rcl_ret_t
 wait_for_service_to_be_ready(
```

Comment:
This function is repeated also here: https://github.com/ros2/rcl/blob/65feb5c7255a41da0311664fbb8104a2f086af2e/rcl/test/rcl/service_fixture.cpp#L32.

It would be good to move it to a common header in `rcl/test`.
Though, it may be out of the scope of the PR.

Reply:
Yes. Good suggestion.   
Update this.